main.go

```diff
@@ -214,34 +210,46 @@ func setupControllers(ctx context.Context, mgr ctrl.Manager, cCache *cache.Cache
 		os.Exit(1)
 	}
 
-	crds := findCustomResources(ctx, mgr)
-
 	opts := []jobframework.Option{
 		jobframework.WithManageJobsWithoutQueueName(manageJobsWithoutQueueName),
 		jobframework.WithWaitForPodsReady(waitForPodsReady(cfg)),
 		jobframework.WithKubeServerVersion(serverVersionFetcher),
 	}
 	err := jobframework.ForEachIntegration(func(name string, cb jobframework.IntegrationCallbacks) error {
 		log := setupLog.WithValues("jobFrameworkName", name)
-		if isFrameworkEnabled(cfg, name) && crds.Has(name) {
-			if err := cb.NewReconciler(
-				mgr.GetClient(),
-				mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", name, constants.KueueName)),
-				opts...,
-			).SetupWithManager(mgr); err != nil {
-				log.Error(err, "unable to create controller")
-				return err
-			}
-			if err := cb.SetupWebhook(mgr, opts...); err != nil {
-				log.Error(err, "Unable to create webhook")
-				return err
-			}
-		} else {
-			if err := noop.SetupWebhook(mgr, cb.JobType); err != nil {
-				log.Error(err, "Unable to create noop webhook")
-				return err
+
+		if isFrameworkEnabled(cfg, name) {
+			if _, err := mgr.GetRESTMapper().RESTMapping(cb.GVK.GroupKind(), cb.GVK.Version); err != nil {
```

Comment:
Could we use something like 
```go
		mgr.GetScheme().ObjectKinds(cb.JobType)	
```
or [apiutil.GVKForObject()](https://github.com/kubernetes-sigs/controller-runtime/blob/40203bfeac472914f36ea35bf386326814f6f82e/pkg/client/apiutil/apimachinery.go#L113) insted if chaging `IntegrationCallbacks `

Reply:
Let me check.

Reply:
As a result, both ways (`mgr.GetScheme().ObjectKinds(cb.JobType) `and `apiutil.GVKForObject()`) are different from what is expected.

As I checked both ways in the KinD cluster, both ways returned the GVK even if the jobFramework CRD wasn't deployed because I guess that we registered all jobFramework schemes as an `init()`.

Also, `mgr.GetScheme().ObjectKinds` returns `false` as a second returned value regardless of whether the jobFramework CRD is deployed in the cluster.

Reply:
Sure, but what I meant is that instead of changing the `IntegrationCallbacks` structure and add new callback in all the integrations, get the GVK with one of those two calls. something like:

```suggestion
			gvk := apiutil.GVKForObject(cb.JobType)
			if _, err := mgr.GetRESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version); err != nil {
```

Reply:
I see.

Reply:
It would work well.

Reply:
Done.